stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/UserReactionAdapter.kt

```diff
@@ -0,0 +1,82 @@
+package io.getstream.chat.android.ui.messages.reactions
+
+import android.view.ViewGroup
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.databinding.StreamItemUserReactionBinding
+import io.getstream.chat.android.ui.utils.extensions.context
+import io.getstream.chat.android.ui.utils.extensions.getDimension
+import io.getstream.chat.android.ui.utils.extensions.inflater
+import io.getstream.chat.android.ui.utils.extensions.updateConstraints
+
+internal class UserReactionAdapter(
+    private val reactionClickListener: ReactionClickListener
+) : ListAdapter<ReactionItem, UserReactionAdapter.UserReactionViewHolder>(ReactionItemDiffCallback()) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserReactionViewHolder {
+        return StreamItemUserReactionBinding
+            .inflate(parent.context.inflater, parent, false)
+            .let { UserReactionViewHolder(it, reactionClickListener) }
+    }
+
+    override fun onBindViewHolder(holder: UserReactionViewHolder, position: Int) {
+        return holder.bind(getItem(position))
+    }
+
+    class UserReactionViewHolder(
+        private val binding: StreamItemUserReactionBinding,
+        private val reactionClickListener: ReactionClickListener
+    ) : RecyclerView.ViewHolder(binding.root) {
+
+        private lateinit var reactionItem: ReactionItem
+
+        init {
+            binding.root.setOnClickListener {
+                reactionClickListener.onReactionClick(reactionItem.reaction)
+            }
+        }
+
+        fun bind(reaction: ReactionItem) {
+            this.reactionItem = reaction
+            bindUserAvatar()
+            bindUserName()
+            bindUserReaction()
+        }
+
+        private fun bindUserAvatar() {
+            val user = reactionItem.reaction.user
+            binding.avatarView.setUserData(user ?: User())
```

Comment:
Can this happen, is there a chance that a reaction belongs to no user? If it can, shouldn't we filter out these reactions before we set them into the adapter?

Reply:
Good idea, I think reaction without user is an edge case. Maybe this can happen when user was deleted?

Reply:
According to the discussion I've found, this was fixed on the server side a year ago. Added double-bang operator.

Reply:
Would it be overkill to make the property non-nullable in the `Reaction` class? Sounds like that should also be fine, if it's guaranteed to be non-null on the server side.

Reply:
Added filtering of reactions without user, just in case